python/lsst/daf/butler/registry/interfaces/_database.py

```diff
@@ -1875,6 +1879,25 @@ def query(
             if connection is not self._session_connection:
                 connection.close()
 
+    def _log_query_and_plan(
+        self,
+        connection: sqlalchemy.Connection,
+        sql: sqlalchemy.sql.expression.Executable | sqlalchemy.sql.expression.SelectBase,
+    ) -> None:
+        """Log the given SQL statement and the DB's plan for executing it if
+        the environment variable DAF_BUTLER_DEBUG_QUERIES is set to a truthy
+        value.
+        """
+        if os.environ.get("DAF_BUTLER_DEBUG_QUERIES", False):
```

Comment:
Can we do this once only?

Reply:
It's a dict lookup, the network I/O and disk I/O surrounding it is going to dwarf the execution time by several orders of magnitude.

Also in notebooks it's nice to be able to twiddle the variable after startup.